### PR TITLE
docs(roadmap): create ROADMAP document

### PR DESCRIPTION
**Pull request by dominguezcelada**
## 📝 Summary

Add Roadmap Document.

Fix #70 

## 📚 References:

Inspired by https://github.com/haadcode/roadmap-generator

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+This document describes the current status and the upcoming milestones of the <PROJECT_NAME> project.
+
+## <PROJECT_NAME>
+
+### Milestone Summary
+
+| Status | Milestone | Goals | ETA |
+| :---: | :--- | :---: | :---: |
+| 🚀 | **[Milestone 1](#milestone-1)** | 0 / 6 | Sat Oct 01 2020 |
+| 🚀 | **[Milestone 2](#milestone-2)** | 0 / 1 | Tue Nov 01 2020 |
```
Comment (dominguezcelada) [inline ROADMAP.md:12]: Let's discuss which are the current milestones (already achieved and the ones in progress). From here, we can brainstorm the following ones.

Once the milestones are defined I can draft the issues and PR's associated to those Milestones

Comment (shreyas44) [reply ROADMAP.md]: 1. Make a list of all Features
2. Build the backend database to incorporate all those features
3. Build the API
4. Build the Front End for Desktop
5. Expand to Mobile

Comment (shreyas44) [reply ROADMAP.md]: @dominguezcelada sorry for the late reply. Do the above 5 milestones sound good?
Also, should we design mobile first or desktop first?

Comment (dominguezcelada) [reply ROADMAP.md]: > 1. Create docker image for the front end and backend using Nginx Unit
> 2. Make a list of all Features
> 3. Build the backend database to incorporate all those features
> 4. Build the API
> 5. Build the Front End for Desktop
> 6. Expand to Mobile
> 7. Deploy the App to AWS

These points sound like technical solutions. Roadmap should be more oriented to features and from there, the tech solutions would be *how we cover those features*

Comment (shreyas44) [reply ROADMAP.md]: Could you give an example?

Comment (dominguezcelada) [reply ROADMAP.md]: - A donnor can:
   - Login/Logout
   - Follow NGOs
   - Comment a post of an NGO
   - Like a post of an NGO
   - etc.
- A NGO can:
   - Login/Logut
   - Post

etc...

---

For non-feature specific milestones like the NGINX Unit one I was thinking about a Tech Debt Milestone:

- Tech Debt:
  - Docker + NGINX Unit
  - etc...

---

Let's say, I'm thinking of milestones as features with business value (or value for users). Then we can also have a tech debt one for those which are not directly related to business value but are important for us.

What do you think? @shreyas44

Comment (shreyas44) [reply ROADMAP.md]: Shouldn't the roadmap show what we'll be building in a given timeframe? Like we finish one layer at a time and once we're done with that, move on to the next layer?

Comment (dominguezcelada) [reply ROADMAP.md]: > Shouldn't the roadmap show what we'll be building in a given timeframe? Like we finish one layer at a time and once we're done with that, move on to the next layer?

Yep! What I'm suggesting is to fix the milestones in a product value perspective and, from those milestones, create tasks/issues associated to those milestones. Depending on the number and size of the issues associated to a milestone and its priority (and dependencies between milestones) we can plan.

In any case, I would not overthink it... If you think I'm being too complex... feel free to draft a PR over the template I landed already on master and we tweak it there.

Comment (shreyas44) [reply ROADMAP.md]: Alright then.

So then the Milestones would be

1. Tech Debt
2. Features List
3. Database & API
4. Front End
5. Deployment

??

Comment (dominguezcelada) [reply ROADMAP.md]: > Alright then.
> 
> So then the Milestones would be
> 
> 1. Tech Debt
> 2. Features List
> 3. Database & API
> 4. Front End
> 5. Deployment
> 
> ??

Sounds good to me to start, yep

**Pull request merged**